### PR TITLE
Streaming backup API

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -494,6 +551,19 @@ async def delBackup(self, name):
         '''
         return await self.cell.delBackup(name)
 
+    @adminapi()
+    async def streamBackupArchive(self, name):
```
Comment (invisig0th) [inline synapse/lib/cell.py:555]: typically named iterBackupArchive() but i'd be comfortable with simply getBackupArchive() as well.

Comment (invisig0th) [reply synapse/lib/cell.py:555]: We should also plumb an easy helper for 1-shot take a backup and stream it ( with optional remove it? )

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -73,6 +76,60 @@ def wrapped(*args, **kwargs):
...
+    Create tarball and stream bytes.
+
+    Args:
+        path (str): Path to the backup.
+
+    Yields:
+        (bytes): File bytes
+    '''
+    output_filename = path + '.tar.gz'
+    link0, file1 = await s_link.linkfile()
```
Comment (invisig0th) [inline synapse/lib/cell.py:90]: oh right!  we have a helper for that! :D

File: synapse/lib/cell.py (hunk 1 above)
Comment (Cisphyx) [reply synapse/lib/cell.py]: Adjusted the naming and added a 1-shot helper

File: synapse/lib/cell.py [hunk 3]
```diff
@@ -1185,6 +1273,102 @@ def walkpath(path):
...
+
+            mesg = repr(e)
+            raise
+
+        finally:
+            raise s_exc.DmonSpawn(mesg=mesg)
+
+    async def iterNewBackupArchive(self, name, user, remove=False):
+
+        await self.runBackup(name)
```
Comment (invisig0th) [inline synapse/lib/cell.py:1327]: `name = await self.runBackup(name)` to allow auto-generation to return a value?

Comment (invisig0th) [reply synapse/lib/cell.py:1327]: Lets make name an optional argument ( just like runBackup() ) to auto-populate a time based name ( here and the telepath endpoint ) and have a test that calls with name=None

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/cell.py (hunk 3 above)
Comment (Cisphyx) [reply synapse/lib/cell.py]: Updated name to an optional arg

**Review by invisig0th — APPROVED** [review]

**Pull request merged**